test/soldiers.test.js

```diff
@@ -0,0 +1,165 @@
+import {
+  afterAll,
+  beforeAll, beforeEach, describe, expect, it, vi,
+} from 'vitest';
+import { client } from '../connections.js';
+import app from '../app.js';
+import * as database from '../database.js';
+
+const { dbName, dbCollection } = database;
+
+beforeAll(async () => {
+  await client.connect();
+});
+
+afterAll(async () => {
+  await client.close();
+});
+
+beforeEach(async () => {
+  await client.db(dbName).collection(dbCollection).deleteMany({});
+});
+
+const testSoldier = {
+  id: '3251084213556',
+  name: 'Shira',
+  rank: '3',
+  limitations: ['none', 'none'],
+};
+
+const testSoldier2 = {
+  id: '3251084213556',
+  name: 'blaaa',
+  rank: '3',
+  limitations: ['none', 'none'],
+};
+
+describe('Post soldier route', () => {
+  it('checks that status is 201 when a soldier was inserted, ann inserted soldier is sent', async () => {
+    const res = await app.inject({
+      method: 'POST',
+      url: '/soldiers',
+      body: testSoldier,
+    });
+    expect(res.statusCode).toBe(201);
+    expect(res.body).toContain('id');
+    expect(res.body).toContain(testSoldier.id);
+    expect(res.body).toContain(testSoldier.name);
+  });
+
+  it('checks that when soldier not created because of duplicate id status is 400', async () => {
+    await app.inject({
+      method: 'POST',
```

Comment:
if you use `POST` to prepare your test, if you have change in the post route you will need to update all the test...
also, if the post not working all the test will failed...

this is not a clear cut because if you use the `database.js` function all those reason still relevant, (and if you just use the db function your test is cupelled to the db, and DRY). I think in your case it reasonably safe to use the `database` function, as an abstraction that is crutial to all the app. if you want we can talk and i will explained more

Reply:
hopefully fixed